[Refactor:Forum] Remove !important from forum.css

sujay-d07 opened the pull request:
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
Closes #11347 
<!-- List issue if it fixes/closes/implements one using the "Fixes #<number>" or "Closes #<number>" syntax -->
forum.css currently contains many instances of !important which forces elements to follow specific styles regardless of other CSS rules.

![image](https://github.com/user-attachments/assets/51394383-fd88-4607-a2e2-214746159ca4)


### What is the new behavior?
Removed all instances of !important from it to follow the Standard Specificity Rules which makes it easier to debug and maintain the code. 

![image](https://github.com/user-attachments/assets/996bed29-7e9c-494e-bd36-026ec277444a)


### Other information?
<!-- Is this a breaking change? -->
<!-- How did you test -->
Added the screenshots to verify that the Forum looks exactly same before and after the changes are made.

shailpatels reviewed (changes requested):
It looks like you went through and removed all the `!important` properties, have you verified the site still looks visually the same?

JManion32 reviewed (changes requested):
Everything appears to be visually correct in light mode, and dark mode. With that being said, I merged the main branch into this PR because there is an instance of !important for dark mode category filters that was added after this was created. I just tested it to see if removing !important would fix it, but it looks like the text colors are still being overridden elsewhere.

**With !important**
![with important](https://github.com/user-attachments/assets/d16438e9-361f-4359-ab01-c9672950256e)

**Without !important (tested fix)**
![no important](https://github.com/user-attachments/assets/1d7c5b41-b18d-416c-ac36-434079553b2c)

Once we identify and fix the source of the override, this PR will be ready to merge into main.

sujay-d07 commented:
> Everything appears to be visually correct in light mode, and dark mode. With that being said, I merged the main branch into this PR because there is an instance of !important for dark mode category filters that was added after this was created. I just tested it to see if removing !important would fix it, but it looks like the text colors are still being overridden elsewhere.

Can you tell me how can I help to fix this and in which PR was the new instances of !important added for dark mode category filters?

sujay-d07 commented:
@JManion32 I have fixed the new issue. Removed the new instance of !important in the dark theme and also made sure that the category buttons are displayed with the correct colours.

with !important:
![Screenshot from 2025-01-25 11-29-37](https://github.com/user-attachments/assets/37afaf50-8360-4e6f-8c2a-dc5630287f99)

without !important + new changes to ForumBar.twig:
![Screenshot from 2025-01-25 11-31-13](https://github.com/user-attachments/assets/74304c9b-3009-4430-8e65-835d27e6a6a5)

JManion32 reviewed (changes requested):
Your changes worked for dark mode because you set all category filters to have white text. However, doing this caused the text to not be visible in light mode:
![sujay_light](https://github.com/user-attachments/assets/f4e712e3-b976-412e-9df2-7d7e13228122)

To fix this, I suggest removing line 128 in ForumBar.twig altogether rather than forcing white text. This will allow colors.css to specifiy if the text color should be the same as the category in light mode, or white in dark mode

With suggested changes in light mode:
![sujay_correction_light](https://github.com/user-attachments/assets/918f09aa-7189-45d6-b2eb-412622826a03)

With suggested changes in dark mode:
![sujay_dark_corrected](https://github.com/user-attachments/assets/dc71bd67-e376-45a9-81c7-c319206ebdc9)

I additionally verified that this PR works on Firefox, and Microsoft Edge since it covers a significant portion of the discussion forum's CSS. Once this change is made, it should be all set to merge.

sujay-d07 commented:
@JManion32 okay will add the suggested changes.

JManion32 reviewed (changes requested):
Something I did not notice from my original review is that the text on the category buttons in light mode did not match the border.
**Suggested Change:**
![suggested change](https://github.com/user-attachments/assets/a3ee331b-cdb6-47dd-b2ee-6fa339eae51f)

**How it should look:**
![how it should look](https://github.com/user-attachments/assets/530a7300-c8f3-4705-ac97-fc241fac255b)

The problem here is that the forum.css file is not able to get each individual category's color, so it has to be set in the .twig file. Implementing the below changes will fix the colors, and make the PR ready for merge.
![Screenshot 2025-01-28 111403](https://github.com/user-attachments/assets/38bf8536-3ae8-40ae-85c4-9764cba0f295)

sujay-d07 commented:
Added the suggested changes of specifying the colors in the ForumBar.twig file.
a) Light Mode:
Before:
![Screenshot from 2025-02-01 12-20-59](https://github.com/user-attachments/assets/8af3ced4-0570-4c85-8654-d7bfde9a8cf1)

After:
![Screenshot from 2025-02-01 12-23-01](https://github.com/user-attachments/assets/19af491f-837d-4c58-8a0a-97b294fc06ab)

b) Dark Mode:
Before:
![Screenshot from 2025-02-01 12-21-35](https://github.com/user-attachments/assets/13470237-7520-4fb2-9b61-f49bc7f20082)

After:
![Screenshot from 2025-02-01 12-23-17](https://github.com/user-attachments/assets/19d83bf0-afba-47a3-880e-545d0829e6b9)

JManion32 reviewed (dismissed):
I have verified that the changes work and everything looks good to me. !important is no longer used and the forum is visually the same. Great work! I will now pass along to other reviewers to ensure that it works on other devices.

sujay-d07 commented:
> I have verified that the changes work and everything looks good to me. !important is no longer used and the forum is visually the same. Great work! I will now pass along to other reviewers to ensure that it works on other devices.

Alright, Thanks!

max1dyer reviewed (changes requested):
All instances of !important are removed in forum.css but there are a few slight visual changes in the discussion forum as a result of this. I tested this on firefox in both light and dark mode.

The header colors for the unread and deleted threads in dark mode are different colors and the light mode blues look different. I believe the unread deleted forum colors are also different for dark mode. 
Before:
![image](https://github.com/user-attachments/assets/5cc9f51e-9065-4e03-be3a-20acfadcb46f)
![image](https://github.com/user-attachments/assets/cf358feb-cdf1-4871-b6b9-1d63d514d78f)

With changes:
![image](https://github.com/user-attachments/assets/934d4c28-8986-4dce-8c4d-0c6414c66e8a)
![image](https://github.com/user-attachments/assets/ab9fcd53-86ef-48f2-bd44-db16db0a0bd8)

The icons for pin and bookmark are now dark grey/black in dark mode.
Before:
![image](https://github.com/user-attachments/assets/4570d9fc-e2cd-4fee-932e-981ee4f28a45)
With changes:
![image](https://github.com/user-attachments/assets/b5c9895b-ba5b-49b0-a792-8e9219bc06a9)

All other visuals look consistent but I'm not sure how to test the remove announcement. One really small change is also line 128 in ForumBar.twig doesnt need a space between color and :. Could you also rename the branch to describe the change?

sujay-d07 commented:
@max1dyer Thanks for pointing these out! I will look into these and fix them👍

JManion32 reviewed (changes requested):
Due to several recent PRs in the forum (namely, removing hardcoded colors in forum.css), it seems as though merging main into here has broken it. As Max said, line 128 in ForumBar.twig should not have a space between color and ":". In addition, some of the colors look off. This is because they are being overridden before the forum.css file. I tested this to ensure it was the case by adding "!important" to the affected code, and sure enough the page looked visually the same again. To fix this issue, you will have to find where the styling of these objects is being overridden. All code below is found in forum.css:

**Bookmark / Pin Button:**
![golden hov](https://github.com/user-attachments/assets/e393784b-2bed-4ec3-92ac-2552c35be584)

**Ellipsis (in the lower right corner of a thread, had the same issue as the pin and bookmark)**
![ellipi](https://github.com/user-attachments/assets/d021d7a3-a750-4502-a131-8d35df06080d)

**Deleted / Unviewed in Light Mode**
![deleted unviewed](https://github.com/user-attachments/assets/dd3b355c-601c-4ef6-be23-57c60d52cdbe)

You will need to find where this variable is getting overridden and ensure that it it being set in forum.css.

By finding where these variables are being overridden and leaving it to forum.css to decide, this PR should be well polished and ready for more reviews.

max1dyer commented on site/app/templates/forum/ForumBar.twig:
```diff
@@ -125,12 +125,20 @@
 			{% for category in categories %}
 				{{ "#categoryid_" ~ category.category_id }}.filter-inactive{
-					color: {{ category.color }};
+					color : {{ category.color }};
```
can we remove the space inserted here

JManion32 reviewed (approved):
The use of !important has been eliminated in the forum and it looks visually the same.

max1dyer reviewed (approved):
The icons are all the right color now. Deleted and unviewed threads have the correct colors. All styles remain consistent.

![image](https://github.com/user-attachments/assets/60aea972-7acf-403f-8d85-e0535642cbfd)
![image](https://github.com/user-attachments/assets/ea748812-3752-4168-aa9e-b6d3849e2e97)
![image](https://github.com/user-attachments/assets/d8dcfa20-4d20-408a-b596-3547c0642cc6)
![image](https://github.com/user-attachments/assets/932b0bff-5c22-414b-bfcf-7feff0227458)
Context from top to bottom: deleted and unviewed, deleted, unviewed, viewed
